Code/GraphMol/Substruct/SubstructMatch.cpp

```diff
@@ -201,13 +203,22 @@ namespace RDKit{
   {
 
     //std::cerr<<"begin match"<<std::endl;
+#ifdef RDK_THREADSAFE_SSS    
```

Comment:
This current strategy eliminates a conditional converts "is recursion Possible" to "did recursion actually happen".  If the code is cleaner with both checks, I'll add it back.